scripts/prepare-preview-builds.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# This script prepares a package to be published as a preview build
+# to GitHub Packages.
+
+if [[ $# -eq 0 ]]; then
+  echo "Missing commit hash."
+  exit 1
+fi
+
+# We use the short commit hash as the prerelease version. This ensures each
+# preview build is unique and can be linked to a specific commit.
+shorthash="$1"
+
+prepare-preview-manifest() {
+  local manifest_file="$1"
+
+  # jq does not support in-place modification of files, so a temporary file is
+  # used to store the result of the operation. The original file is then
+  # overwritten with the temporary file.
+  jq --raw-output --arg hash "$shorthash" --from-file scripts/prepare-preview-builds.jq "$manifest_file" > temp.json
+  mv temp.json "$manifest_file"
+}
+
+echo "Preparing manifests..."
+while IFS=$'\t' read -r location name; do
+  echo "- $name"
+  prepare-preview-manifest "$location/package.json"
+done < <(yarn workspaces list --no-private --json | jq --slurp --raw-output 'map(select(.location != ".")) | map([.location, .name]) | map(@tsv) | .[]')
+
+echo "Installing dependencies..."
+yarn install --no-immutable
```

Comment:
Yarn will use `--immutable` by default when it detects it's being run in a CI environment. We have to turn this off because we have to regenerate the lockfile (since we are changing the names of packages etc.). If we don't do this then we won't be able to run any Yarn commands following this script.